### PR TITLE
Remove max delegate interest for witnesses #1008

### PR DESCRIPTION
**Pull request by maslenitsa93**
#1008

### CODE REVIEW

File: libraries/chain/chain_properties_evaluators.cpp [hunk 1]
```diff
@@ -52,6 +52,7 @@ namespace golos { namespace chain {
 
         result_type operator()(const chain_properties_19& p) const {
             ASSERT_REQ_HF(STEEMIT_HARDFORK_0_19__295, "chain_properties_19");
+            GOLOS_CHECK_VALUE_LE(p.max_delegated_vesting_interest_rate, 80*STEEMIT_1_PERCENT);
```
Comment (afalaleev) [inline libraries/chain/chain_properties_evaluators.cpp]: 1. After HF19 the previous assert is not required - remove it.
2. Use `STEEMIT_MAX_DELEGATED_VESTING_INTEREST_RATE_PRE_HF21` - see the next comment.
3. This line should be protected by checking on STEEMIT_HARDFORK_0_21__1008: 
```
if (!has_hardfork(STEEMIT_HARDFORK_0_21__1008)) {
    GOLOS_CHECK_VALUE_LE(p.max_delegated_vesting_interest_rate, STEEMIT_MAX_DELEGATED_VESTING_INTEREST_RATE_PRE_HF21);
}
```

File: libraries/protocol/include/golos/protocol/config.hpp [hunk 2]
```diff
@@ -135,7 +135,6 @@
 #define GOLOS_MIN_DELEGATION_MULTIPLIER             10
 
 #define STEEMIT_DEFAULT_DELEGATED_VESTING_INTEREST_RATE (25*STEEMIT_1_PERCENT) // 25%
-#define STEEMIT_MAX_DELEGATED_VESTING_INTEREST_RATE     (80*STEEMIT_1_PERCENT) // 80%
```
Comment (afalaleev) [inline libraries/protocol/include/golos/protocol/config.hpp]: Rename to `STEEMIT_MAX_DELEGATED_VESTING_INTEREST_RATE_PRE_HF19`
Add a new macro `#define STEEMIT_MAX_DELEGATED_VESTING_INTEREST_RATE STEEMIT_100_PERCENT`

File: libraries/protocol/steem_operations.cpp [hunk 3]
```diff
@@ -298,7 +298,7 @@ namespace golos { namespace protocol {
             GOLOS_CHECK_VALUE_LEGE(comments_per_window, 1, comments_window);
             GOLOS_CHECK_VALUE_LEGE(votes_window, 1, std::numeric_limits<uint16_t>::max() / 2);
             GOLOS_CHECK_VALUE_LEGE(votes_per_window, 1, votes_window);
-            GOLOS_CHECK_VALUE_LE(max_delegated_vesting_interest_rate, STEEMIT_MAX_DELEGATED_VESTING_INTEREST_RATE);
+            GOLOS_CHECK_VALUE_LE(max_delegated_vesting_interest_rate, STEEMIT_100_PERCENT);
```
Comment (afalaleev) [inline libraries/protocol/steem_operations.cpp]: Use `STEEMIT_MAX_DELEGATED_VESTING_INTEREST_RATE` - see previous comment

File: libraries/chain/chain_properties_evaluators.cpp [hunk 4]
```diff
@@ -50,19 +50,24 @@ namespace golos { namespace chain {
                 : _db(db), _wprops(wprops) {
         }
 
-        result_type operator()(const chain_properties_19& p) const {
-            ASSERT_REQ_HF(STEEMIT_HARDFORK_0_19__295, "chain_properties_19");
-            _wprops = p;
+        result_type operator()(const chain_properties_19& props) const {
+            GOLOS_CHECK_PARAM(props, {
+                if (!_db.has_hardfork(STEEMIT_HARDFORK_0_21__1008)) {
```
Comment (zxcat) [inline libraries/chain/chain_properties_evaluators.cpp]: is there check that `max_delegated_vesting_interest_rate` ≤ 100% ?

**Pull request merged**